### PR TITLE
[bug_5753] Added tests.

### PR DESCRIPTION
**Pull request by pwodzinski**
Added some additional test cases to tests.py
https://bugzilla.stxnext.pl/show_bug.cgi?id=5753#c2

### CODE REVIEW

**Comment by dpolcyn** [comment]
Move tests to ```/src/presence_analyzer/tests.py```. Don't create new file containing tests.

File: src/presence_analyzer/tests.py [hunk 1]
```diff
@@ -54,6 +54,31 @@ def test_api_users(self):
         self.assertDictEqual(data[0], {u'user_id': 10, u'name': u'User 10'})
 
 
+# Bellow i add my own tests
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: This comment isn't needed here. Delete.

File: src/presence_analyzer/tests.py [hunk 2]
```diff
@@ -54,6 +54,31 @@ def test_api_users(self):
...
+
+    def test_own(self):
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Add docstring.

File: src/presence_analyzer/tests.py [hunk 3]
```diff
@@ -54,6 +54,31 @@ def test_api_users(self):
...
+        resp = self.client.get('/nnn')
+        self.assertEqual(resp.status_code, 404)
+
+    def test_next(self):
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ditto

File: src/presence_analyzer/tests.py [hunk 4]
```diff
@@ -54,6 +54,31 @@ def test_api_users(self):
...
+        resp = self.client.get('/api/v1/users')
+        data = json.loads(resp.data)
+        self.assertNotEqual(len(data), 3)
+
+    def test_apiNow(self):
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ditto

File: src/presence_analyzer/tests.py [hunk 5]
```diff
@@ -54,6 +54,31 @@ def test_api_users(self):
...
+        resp = self.client.get('/api/v1/mean_time_weekday/10')
+        self.assertEqual(resp.status_code, 200)
+        resp = self.client.get('/api/v1/mean_time_weekday/101')
+        self.assertEqual(resp.status_code, 404)
+
+    def test_weekday(self):
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ditto

File: src/presence_analyzer/tests.py (hunk 2 above)
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: This function name doesn't mean which one function you want to test. Fix it.

File: src/presence_analyzer/tests.py (hunk 3 above)
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ditto

File: src/presence_analyzer/tests.py (hunk 4 above)
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ditto

File: src/presence_analyzer/tests.py (hunk 5 above)
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ditto

**Comment by dpolcyn** [comment]
Change pull request name:
Example: 
```
[bug_xxxx] Added tests.

**Comment by dpolcyn** [comment]
Please squash this 2 commits.

**Comment by dpolcyn** [comment]
Add pep8 result and tests result.

**Comment by dpolcyn** [comment]
Add link to bugzilla.

**Comment by dpolcyn** [comment]
You missed tests for functions in utils.py

File: src/presence_analyzer/tests.py [hunk 6]
```diff
@@ -54,6 +54,31 @@ def test_api_users(self):
...
+        resp = self.client.get('/api/v1/presence_weekday/10')
+        data = json.loads(resp.data)
+        self.assertEqual(len(data), 8)
+        self.assertEqual(data[0], [u'Weekday', u'Presence (s)'])
+        resp = self.client.get('/api/v1/presence_weekday/12')
+        self.assertEqual(resp.status_code, 404)
+
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: One more enter here.

File: src/presence_analyzer/tests.py [hunk 7]
```diff
@@ -6,15 +6,15 @@
 import json
 import datetime
 import unittest
+import sys
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: You don't need this import.

File: src/presence_analyzer/tests.py [hunk 8]
```diff
@@ -6,15 +6,15 @@
...
 
 from presence_analyzer import main, views, utils
 
 
 TEST_DATA_CSV = os.path.join(
-    os.path.dirname(__file__), '..', '..', 'runtime', 'data', 'test_data.csv'
+    os.path.dirname(__file__), '..', '..','runtime', 'data', 'test_data.csv'
 )
 
-
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: One more enter here.

File: src/presence_analyzer/tests.py [hunk 9]
```diff
@@ -6,15 +6,15 @@
...
 import datetime
 import unittest
+import sys
 
 from presence_analyzer import main, views, utils
 
 
 TEST_DATA_CSV = os.path.join(
-    os.path.dirname(__file__), '..', '..', 'runtime', 'data', 'test_data.csv'
+    os.path.dirname(__file__), '..', '..','runtime', 'data', 'test_data.csv'
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ```os.path.dirname(__file__), '..', '..', 'runtime', 'data', 'test_data.csv'```

File: src/presence_analyzer/tests.py [hunk 10]
```diff
@@ -54,6 +54,31 @@ def test_api_users(self):
         self.assertDictEqual(data[0], {u'user_id': 10, u'name': u'User 10'})
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Too many black lines.

File: src/presence_analyzer/tests.py [hunk 11]
```diff
@@ -54,6 +54,31 @@ def test_api_users(self):
...
 
 
+# Bellow i add my own tests
+
+    def test_own(self):
+        resp = self.client.get('/nnn')
+        self.assertEqual(resp.status_code, 404)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Move this case to ```test_mainpage```

File: src/presence_analyzer/tests.py [hunk 12]
```diff
@@ -54,6 +54,31 @@ def test_api_users(self):
...
+
+    def test_next(self):
+        resp = self.client.get('/api/v1/users')
+        data = json.loads(resp.data)
+        self.assertNotEqual(len(data), 3)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Move this case to ```test_api_users```

File: src/presence_analyzer/tests.py [hunk 13]
```diff
@@ -54,6 +54,31 @@ def test_api_users(self):
...
+
+    def test_apiNow(self):
+        resp = self.client.get('/api/v1/mean_time_weekday/10')
+        self.assertEqual(resp.status_code, 200)
+        resp = self.client.get('/api/v1/mean_time_weekday/101')
+        self.assertEqual(resp.status_code, 404)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Please check if resp.content_type is valid.

File: src/presence_analyzer/tests.py [hunk 14]
```diff
@@ -54,6 +54,31 @@ def test_api_users(self):
...
+
+    def test_weekday(self):
+        resp = self.client.get('/api/v1/presence_weekday/10')
+        data = json.loads(resp.data)
+        self.assertEqual(len(data), 8)
+        self.assertEqual(data[0], [u'Weekday', u'Presence (s)'])
+        resp = self.client.get('/api/v1/presence_weekday/12')
+        self.assertEqual(resp.status_code, 404)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Please check if ```resp.content_type``` is valid.

File: src/presence_analyzer/tests.py [hunk 15]
```diff
@@ -54,6 +54,31 @@ def test_api_users(self):
...
+        resp = self.client.get('/api/v1/mean_time_weekday/10')
+        self.assertEqual(resp.status_code, 200)
+        resp = self.client.get('/api/v1/mean_time_weekday/101')
+        self.assertEqual(resp.status_code, 404)
+
+    def test_weekday(self):
+        resp = self.client.get('/api/v1/presence_weekday/10')
+        data = json.loads(resp.data)
+        self.assertEqual(len(data), 8)
+        self.assertEqual(data[0], [u'Weekday', u'Presence (s)'])
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Enter after this line

Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Please check full list.

File: src/presence_analyzer/tests.py (hunk 13 above)
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: You have to check if ```resp.data``` is valid.

File: src/presence_analyzer/tests.py [hunk 16]
```diff
@@ -51,7 +53,31 @@ def test_api_users(self):
...
+        self.assertEqual(resp.content_type, 'application/json')
+        self.assertEqual(data[0], [u'Weekday', u'Presence (s)'])
+        resp = self.client.get('/api/v1/presence_weekday/12')
+        self.assertEqual(resp.status_code, 404)
+
+    def test_time_weekday(self):
+        resp = views.mean_time_weekday_view(10)
+        data = json.loads(resp.data)
+        self.assertEqual(resp.content_type, 'application/json')
+        self.assertEqual(data[1], [u'Tue', 30047.0])
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Check more cases or/and full list.

File: src/presence_analyzer/tests.py [hunk 17]
```diff
@@ -51,7 +53,31 @@ def test_api_users(self):
...
+
+    def test_presence_weekday_view(self):
+        resp = views.presence_weekday_view(10)
+        data = json.loads(resp.data)
+        self.assertEqual(resp.content_type, 'application/json')
+        self.assertEqual(data[3], [u'Wed', 24465])
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ditto

**Comment by dpolcyn** [comment]
Please add description.

File: src/presence_analyzer/tests.py [hunk 18]
```diff
@@ -86,6 +120,59 @@ def test_get_data(self):
...
+            30047
+        )
+        self.assertEqual(len(data), 2)
+
+    def test_mean(self):
+        self.assertEqual(utils.mean([1, 2, 3, 4]), 2.5)
+        self.assertEqual(utils.mean([]), 0)
+        self.assertEqual(utils.mean([-1]), -1.0)
+        self.assertEqual(utils.mean([0]), 0.0)
+        self.assertEqual(utils.mean([146, 2346, 34563.1, 4345]), 10350.025)
```
Comment (mjelonek92) [inline src/presence_analyzer/tests.py]: It's a good practice to use ```assertAlmostEqual``` to compare floats.

File: src/presence_analyzer/tests.py [hunk 19]
```diff
@@ -86,6 +120,59 @@ def test_get_data(self):
...
+        weekdays = utils.group_by_weekday(data[10])
+        self.assertIsInstance(weekdays, list)
+        self.assertIn([30047], weekdays)
+        self.assertEqual(len(weekdays), 7)
+
+    def test_seconds_since(self):
+        data = utils.get_data()
+        sample_date = datetime.date(2013, 9, 10)
+        get_time = data[10][sample_date]['start']
+        seconds = utils.seconds_since_midnight(get_time)
```
Comment (mjelonek92) [inline src/presence_analyzer/tests.py]: Unused?

File: src/presence_analyzer/tests.py [hunk 20]
```diff
@@ -86,6 +120,59 @@ def test_get_data(self):
...
+            86399
+        )
+
+    def test_interval(self):
+        data = utils.get_data()
+        sample_date = datetime.date(2013, 9, 10)
+        get_time_start = data[10][sample_date]['start']
+        get_time_end = data[10][sample_date]['end']
+        self.assertEqual(utils.interval(
+            get_time_start, get_time_end),
```
Comment (mjelonek92) [inline src/presence_analyzer/tests.py]: Perhaps?
```python
self.assertEqual(
    utils.interval(...),
    30047
)
```

File: src/presence_analyzer/tests.py [hunk 21]
```diff
@@ -53,6 +52,41 @@ def test_api_users(self):
         self.assertEqual(len(data), 2)
         self.assertDictEqual(data[0], {u'user_id': 10, u'name': u'User 10'})
 
+    def test_time_weekday(self):
+        resp = self.client.get('/api/v1/mean_time_weekday/11')
+        self.assertEqual(resp.status_code, 200)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: You should also test status code for not existing ids.

File: src/presence_analyzer/tests.py [hunk 22]
```diff
@@ -53,6 +52,41 @@ def test_api_users(self):
...
+                [u'Thu', 22984.0],
+                [u'Fri', 6426.0],
+                [u'Sat', 0],
+                [u'Sun', 0]
+            ]
+        )
+
+    def test_presence_weekday_view(self):
+        resp = self.client.get('/api/v1/presence_weekday/10')
+        self.assertEqual(resp.status_code, 200)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ditto

**Pull request merged**